opencti-platform/opencti-front/src/private/components/analyses/external_references/ExternalReferenceFileImportViewer.tsx

```diff
@@ -40,9 +41,32 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-const importValidation = (t: (value: string) => string) => Yup.object().shape({
-  connector_id: Yup.string().required(t('This field is required')),
-});
+interface ConnectorConfiguration {
+  configuration: string;
+  id: string;
+  name: string;
+}
+interface Connector {
+  id: string;
+  name: string;
+  active: boolean;
+  connector_scope: string[];
+  updated_at: string;
+  configurations: ConnectorConfiguration[];
+}
+
+const importValidation = (t: (value: string) => string, configurations: boolean) => {
```

Comment:
Maybe you can create a utils files in order to define the function. I think you are using it multiple time.